dev-guide/api-conventions.md

```diff
@@ -279,6 +279,37 @@ myPlatformConfig:
 
 This is invalid. Multiple structs have been configured and no discriminant is provided.
 
+### Prefer consistency with Kubernetes style APIs
+
+As both Kubernetes and OpenShift have a number of integrations with cloud and infrastructure platforms, there are a
+number of different abstractions built into OpenShift that abstract and extend infrastructure capabilities.
+For example, the Machine API or platform specific storage integrations within OpenShift.
+
+These integrations often result in adding new API fields to OpenShift to allow users to configure various features of
+the platform. Often, different platform's concepts are similar, but their naming of particular features can be specific
+to their platform.
+If we copy verbatim the API from a platform, this may or may not be compliant with OpenShift conventions and may or
+may not make sense in a wider scope.
+
+If we intend to support similar features across platforms, it is preferable to have similar APIs for these features
+within OpenShift rather than mimicking the upstream API. This has the benefit of providing consistency to users when
+using OpenShift across multiple platforms and reducing the learning curve when installing OpenShift on a new platform.
+Where possible, we should link to the platform specific documentation for features in the description of our own APIs.
+
+While we appreciate that reusing the platform specific terminology can make it easier for someone familiar with the
+platform to understand the API, we prefer to stick to Kubernetes style conventions (eg preferring PascalCase for
+enumerated values) when abstracting platform specific APIs as this allows us to build a consistent looking API across
+the OpenShift product.
+Differences between our APIs and platform APIs can be handled within the controller backing the API.
+
+We have seen examples in the past where platform APIs are not intuitive to the value to the end user.
```

Comment:
I'm having a little trouble parsing "not intuitive to the value to the end user". Is that the value of an API parameter or the benefit to the user of an API?

Reply:
Benefit is a better word for what I meant yes. Is this easier to understand if I switch `value to the end user` to `benefit for the end user`?

Reply:
Is it that the platform API is not intuitive or that the benefit of the API is not intuitive?

How about this?

```suggestion
We have seen examples in the past where platform APIs are not intuitive to the end user.
```

Reply:
Or maybe

```suggestion
We have seen examples in the past where the value of platform APIs is not intuitive to the end user.
```